src/main/java/fi/okm/jod/yksilo/controller/KeskusteluController.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2024 The Finnish Ministry of Education and Culture, The Finnish
+ * The Ministry of Economic Affairs and Employment, The Finnish National Agency of
+ * Education (Opetushallitus) and The Finnish Development and Administration centre
+ * for ELY Centres and TE Offices (KEHA).
+ *
+ * Licensed under the EUPL-1.2-or-later.
+ */
+
+package fi.okm.jod.yksilo.controller;
+
+import fi.okm.jod.yksilo.domain.LocalizedString;
+import fi.okm.jod.yksilo.service.inference.InferenceService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import java.net.URI;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping(path = "/api/keskustelut")
+@Slf4j
+@Tag(name = "keskustelut", description = "Keskustelut (POC)")
+public class KeskusteluController {
+
+  private final InferenceService<Request, Response> inferenceService;
+  private final String endpoint;
+
+  public KeskusteluController(
+      InferenceService<Request, Response> inferenceService,
+      @Value("${jod.keskustelu.endpoint}") String endpoint) {
+    this.inferenceService = inferenceService;
+    this.endpoint = endpoint;
+  }
+
+  @PostMapping
+  @Operation(summary = "Creates or continues a keskustelu")
+  public ResponseEntity<Response> createOrContinueKeskustelu(
+      @RequestHeader(value = "X-Session-Id", required = false) String sessionId,
```

Comment:
Perhaps it would be better to keep the session id (or keskusteluId) as part of the request and response bodies? We can later consider modeling this as a rest resource (POST createKeskustelu would create a new keskustelu, and POST keskustelu/{id} would continue an existing one).